binderhub/app.py

```diff
@@ -109,17 +109,24 @@ class BinderHub(Application):
         config=True
     )
 
-    hub_login_url = Unicode(
-        None,
-        allow_none=True,
+    hub_api_token = Unicode(
+        help="""API token for talking to the JupyterHub API""",
+        config=True,
+    )
+    hub_url = Unicode(
         help="""
-        The hub login URL to redirect the user to after image is built.
+        The base URL of the JupyterHub instance where users will run.
 
-        It'll get all runtime parameters as query parameters, and it is the responsibility
-        of the hub to launch this properly
+        Must end in '/'
+        e.g. https://hub.mybinder.org/
         """,
-        config=True
+        config=True,
     )
+    @validate('hub_url')
+    def _add_slash(self, proposal):
```

Comment:
@minrk `proposal` is a bit vague. A docstring would help clarify.

Reply:
docstring added. This is a traitlets validator, which always have this signature.